src/Sensor.cc

```diff
@@ -196,6 +213,56 @@ bool SensorPrivate::SetTopic(const std::string &_topic)
   return true;
 }
 
+//////////////////////////////////////////////////
+void Sensor::PublishMetrics(const ignition::common::Time &_now)
+{
+  return this->dataPtr->PublishMetrics(_now);
+}
+
+void SensorPrivate::PublishMetrics(const ignition::common::Time &_now)
+{
+  if(!this->performanceSensorMetricsPub)
+  {
+    this->performanceSensorMetricsPub =
+      node.Advertise<msgs::PerformanceSensorMetrics>(
+          this->topic + "/performance_metrics");
```

Comment:
check here if the topic is valid ? `transport::TopicUtils::AsValidTopic` example:

https://github.com/ignitionrobotics/ign-sensors/blob/5935721047857a436e6482031793070605a758b8/src/Sensor.cc#L205-L210

Reply:
Nice call, thanks!
Done.